File: 20210111/멀쩡한 사각형/lololalayoho.cpp
#include<iostream>
using namespace std;

long long gcd(long a, long b){
	long c;
	while (b != 0){
		c = a % b;
		a = b;
		b = c;
	}
	return a;
}

long long solution(int w, int h) {
	long long answer = 1;
	answer = (long)w * h;
	answer -= (long long)(w + h - gcd(w, h));
	return answer;
}

File: 20210111/문자열 압축/lololalayoho.py
def push_word(s,size):
    word = ""
    return word + s[:size]

def solution(s):
    answer = 999999999
    for j,hello in enumerate(s):
        i = j+1

        sentence = ""
        index = 0
        while index+i <= len(s) and int(len(s) - index) >= i:
            word = push_word(s[index:], i)
            cnt = 0
            if s[index:index+i]==word:
                while s[index:index+i]==word:
                    index = index + i
                    cnt = cnt + 1
                if cnt==1:
                    sentence = sentence + word
                else:
                    sentence = sentence + str(cnt) + word
        sentence = sentence + s[index:]
        if len(sentence) < answer:
            answer = len(sentence)
    return answer